util/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2015-2022 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use std::env;
+
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+
+fn get_env() -> String {
+    env::var("RUST_LOG").unwrap_or_default()
+}
+
+fn get_levels<T: ToString>(bin: &str, level: Option<T>) -> String {
+    let trust_dns_crates = level.map(|level| format!(
+        "{bin}={level},trust_dns_util={level},trust_dns_client={level},trust_dns_server={level},trust_dns_proto={level},trust_dns_resolver={level}",
+        bin = bin,
+        level = level.to_string().to_lowercase(),
+    ));
```

Comment:
as i mentioned above, these could be added to the `EnvFilter` programmatically using `add_directive`.

Reply:
yeah, I was thinking about that... what I *really* wanted here was to use this as the "default" directive, but that didn't seem possible ultimately.

Reply:
you mean, if there's no env var set, use this; otherwise, use the env var and throw all of this away? there isn't an API for that, but you could implement it yourself by checking the env var and then parsing it.

we could also add an API to the builder for setting more complicated defaults that are more than a single directive? that could be worth a `tracing` feature request...

Reply:
Yeah, I think the latter will be more desirable. What I have now achieves that goal close enough.

Reply:
great! and, yeah, what you're doing now definitely _works_, it would just be nice if we had better APIs for this use case...